client/lcd/lcd_test.go

```diff
@@ -1043,16 +1047,31 @@ func getDelegatorUnbondingDelegations(t *testing.T, port string, delegatorAddr s
 	return ubds
 }
 
-func getDelegatorRedelegations(t *testing.T, port string, delegatorAddr sdk.AccAddress) []stake.Redelegation {
-	res, body := Request(t, port, "GET", fmt.Sprintf("/stake/delegators/%s/redelegations", delegatorAddr), nil)
-	require.Equal(t, http.StatusOK, res.StatusCode, body)
+func getRedelegations(t *testing.T, port string, delegatorAddr sdk.AccAddress, srcValidatorAddr sdk.ValAddress, dstValidatorAddr sdk.ValAddress) []stake.Redelegation {
+	var res *http.Response
+	var body string
+
+	endpoint := "/stake/redelegations?"
```

Comment:
I still think this is a terrible design, we should only add the query parameters to `/stake/delegators/{delegatorAddr}/redelegations`, not changing the current structure. cc: @faboweb

Reply:
The thing I don't like about `/stake/delegators/{delegatorAddr}/redelegations` is that it's trying to put a fake structure on data that is not actually structured like in the blockchain.  Redelegations are not attributes of a delegator, but rather `delegator`, `srcValidator`, and `dstValidator` are attributes of a `redelegation`.

Reply:
I would agree with @fedekunze here. I don't think how we have the data structured internally matters to users. Presenting the data in a way that is logically consistent (delegators have redelegations) is a great idea.

Reply:
What are /stake/delegators/<>/redelegations vs /stake/validators/<>/redelegations?

I think over-emphasis on the URL design of API endpoints often leads to premature design constraints that end up causing more headache down the line.  It is simpler to model the API after what is modeled in the implementation.

REST isn't just about pretty URLs... there's nothing wrong with query parameters, esp given a context where we will want to provide filters etc.  I favor Sunny's approach here, it's faster to execute and less brittle going into the future.

The end users here are developers who can deal with query parameters.  We're not talking about website URLs, which is a different story.

Reply:
> It is simpler to model the API after what is modeled in the implementation.

I think simple to implement is not what we want with an API. An API should be consistent for quiet some time so developers don't need to adjust to constant changes (this is frustrating from experience). We should come up with a design that reflects the expectation of the developers that want to use the API. Let's invite more of them to stakeholder meetings and discuss together with them the API design.